core/tests/webdriverio_utils/StoryEditorPage.js

```diff
@@ -149,11 +211,143 @@ var StoryEditorPage = function() {
     await action.click('Back to story editor tab', backToStoryEditorButton);
   };
 
+  this.expectChaptersListToBe = async function(chapters) {
+    await this.expectNumberOfChaptersToBe(chapters.length);
+    for (var i = 0; i < chapters.length; i++) {
+      var chapterTitles = await chapterTitlesSelector();
+      expect(await chapterTitles[i].getText()).toEqual(chapters[i]);
+    }
+  };
+
+  this.dragChapterToAnotherChapter = (async function(chapter1, chapter2) {
+    var chapterTitles = await chapterTitlesSelector();
+    await waitFor.visibilityOf(
+      chapterTitles[0],
+      'Chapter titles taking too long to appear.');
+    var matchFound = false;
+    for (var i = 0; i < chapterTitles.length; i++) {
+      if (await chapterTitles[i].getText() === chapter1) {
+        matchFound = true;
+        break;
+      }
+    }
+    expect(matchFound).toBe(true);
+    var toMove = chapterTitles[i];
+
+    matchFound = false;
+    for (var i = 0; i < chapterTitles.length; i++) {
+      if (await chapterTitles[i].getText() === chapter2) {
+        matchFound = true;
+        break;
+      }
+    }
+    expect(matchFound).toBe(true);
+    var target = chapterTitles[i];
+    await general.dragAndDrop(toMove, target);
+  });
+
+  this.expectDestinationToBe = async function(chapterName) {
+    var pattern = '\s*' + chapterName + '\s*';
+    return expect(await nextChapterCard.getText()).toMatch(pattern);
+  };
+
   this.expectNumberOfChaptersToBe = async function(count) {
     var chapterTitles = await chapterTitlesSelector();
-    expect(await chapterTitles.length).toEqual(count);
+    expect(chapterTitles.length).toEqual(count);
+  };
+
+  this.expectNotesToBe = async function(richTextInstructions) {
+    await forms.expectRichText(storyNotes).toMatch(richTextInstructions);
+  };
+
+  this.expectTitleToBe = async function(title) {
+    await waitFor.visibilityOf(
```

Comment:
presenceOf , right? Ditto same thing inside other functions too.

Reply:
As discussed, presenceOf in webdriverio only looks for the element in the visible screen hence if the element is even present on screen but we need to scroll to get to that element this will throw error hence I am using visibilityOf instead.